ComfortZoneApp/app/src/main/java/com/example/comfortzone/ui/HostActivity.java

```diff
@@ -81,30 +88,58 @@ private void initViews() {
         bottomNavigationView = findViewById(R.id.bottom_navigation);
     }
 
+    private void createFragments() {
+        flightFragment = new FlightFragment();
+        inputFragment = new InputFragment();
+        profileFragment = new ProfileFragment();
+    }
+
     private void listenerSetup() {
         bottomNavigationView.setOnItemSelectedListener(new NavigationBarView.OnItemSelectedListener() {
             @Override
             public boolean onNavigationItemSelected(@NonNull MenuItem item) {
-                Fragment fragment;
                 switch (item.getItemId()) {
                     case R.id.action_flight:
-                        fragment = new FlightFragment();
+                        fragment = flightFragment;
+                        getAnimationLeftToRight();
                         break;
                     case R.id.action_input:
-                        fragment = new InputFragment();
+                        if (fragment == flightFragment) {
+                            getAnimationRightToLeft();
+                        } else {
+                            getAnimationLeftToRight();
+                        }
+                        fragment = inputFragment;
                         break;
                     case R.id.action_profile:
                     default:
-                        fragment = new ProfileFragment();
+                        fragment = profileFragment;
+                        getAnimationRightToLeft();
                         break;
                 }
-                fragmentManager.beginTransaction().replace(R.id.flContainer, fragment, "input").commit();
+                fragmentManager.beginTransaction()
+                        .setCustomAnimations(
+                                enterAnimation,
+                                exitAnimation
+                        )
+                        .replace(R.id.flContainer, fragment, "input")
+                        .commit();
                 return true;
             }
         });
         bottomNavigationView.setSelectedItemId(R.id.action_profile);
     }
 
+    private void getAnimationRightToLeft() {
+        enterAnimation = R.anim.enter_from_right;
+        exitAnimation = R.anim.exit_to_left;
```

Comment:
instead of assigning to member variables, which you only used in single place, try returning an int array instead or Pair: https://docs.oracle.com/javase/9/docs/api/javafx/util/Pair.html

otherwise, member variable is ok, but maybe you should prefix the method as `set` instead of `get`